specs/market-api.yaml

```diff
@@ -1128,26 +1114,90 @@ components:
           type: string
           format: date-time
 
+    PropertyQueryEvent:
+      allOf:
+        - $ref: '#/components/schemas/Event'
+        - type: object
+          properties:
+            propertyQuery:
+              $ref: '#/components/schemas/PropertyQuery'
+
     ProposalEvent:
+      allOf:
+        - $ref: '#/components/schemas/Event'
+        - type: object
+      properties:
+        proposal:
+          $ref: '#/components/schemas/Proposal'
+
+    ProposalRejectedEvent:
       allOf:
         - $ref: '#/components/schemas/Event'
         - type: object
           properties:
-            proposal:
-              $ref: '#/components/schemas/Proposal'
+            proposalId:
+              type: string
+              readOnly: true
+            reason:
+              $ref: 'common.yaml#/components/schemas/Reason'
 
-    AgreementEvent:
+    AgreementProposalEvent:
       allOf:
         - $ref: '#/components/schemas/Event'
         - type: object
           properties:
             agreement:
               $ref: '#/components/schemas/Agreement'
 
-    PropertyQueryEvent:
+
+    # note this should be specified as a sub-class of Event, BUT
+    # a number of OpenAPI code generators fail on multistory inheritance hierarchies :(
+    AgreementEvent:
+      type: object
+      required:
+        - eventType
+        - eventDate
+      discriminator:
+        propertyName: eventType
+      properties:
+        eventType:
+          type: string
+        eventDate:
+          type: string
+          format: date-time
+        agreementId:
+          type: string
+
+    AgreementAcceptedEvent:
       allOf:
-        - $ref: '#/components/schemas/Event'
+        - $ref: '#/components/schemas/AgreementEvent'
+        - type: object
+
+    AgreementRejectedEvent:
+      allOf:
+        - $ref: '#/components/schemas/AgreementEvent'
         - type: object
           properties:
-            propertyQuery:
-              $ref: '#/components/schemas/PropertyQuery'
+            reason:
+              $ref: 'common.yaml#/components/schemas/Reason'
+
+    AgreementCancelledEvent:
+      allOf:
+        - $ref: '#/components/schemas/AgreementEvent'
+        - type: object
+          properties:
+            reason:
+              $ref: 'common.yaml#/components/schemas/Reason'
+
+    AgreementTimeoutEvent:
```

Comment:
This event should be covered in the `collectAgreementEvents ` endpoint description also.

Reply:
Agreed.